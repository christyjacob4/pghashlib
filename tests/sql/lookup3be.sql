-- Test basic hash functionality with text
SELECT lookup3be('hello world');

-- Test hash with different text inputs
SELECT lookup3be('test string');
SELECT lookup3be('another test');

-- Test text input with custom initval
SELECT lookup3be('hello world', 42);
SELECT lookup3be('hello world', 123);

-- Test bytea input
SELECT lookup3be('hello world'::bytea);

-- Test bytea input with custom initval
SELECT lookup3be('hello world'::bytea, 42);

-- Test integer input
SELECT lookup3be(12345);
SELECT lookup3be(-12345);

-- Test integer input with custom initval
SELECT lookup3be(12345, 42);
SELECT lookup3be(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT lookup3be('consistent test') = lookup3be('consistent test');

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3be('initval test', 1) != lookup3be('initval test', 2);

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3be'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';