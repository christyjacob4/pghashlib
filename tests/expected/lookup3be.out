-- Test basic hash functionality with text
SELECT lookup3be('hello world');
 lookup3be  
------------
 -942795449
(1 row)

-- Test hash with different text inputs
SELECT lookup3be('test string');
 lookup3be  
------------
 1077573433
(1 row)

SELECT lookup3be('another test');
  lookup3be  
-------------
 -2119669366
(1 row)

-- Test text input with custom initval
SELECT lookup3be('hello world', 42);
 lookup3be  
------------
 1545643001
(1 row)

SELECT lookup3be('hello world', 123);
 lookup3be  
------------
 1709307537
(1 row)

-- Test bytea input
SELECT lookup3be('hello world'::bytea);
 lookup3be  
------------
 -942795449
(1 row)

-- Test bytea input with custom initval
SELECT lookup3be('hello world'::bytea, 42);
 lookup3be  
------------
 1545643001
(1 row)

-- Test integer input
SELECT lookup3be(12345);
 lookup3be  
------------
 -185163110
(1 row)

SELECT lookup3be(-12345);
  lookup3be  
-------------
 -1622048687
(1 row)

-- Test integer input with custom initval
SELECT lookup3be(12345, 42);
 lookup3be  
------------
 1051714761
(1 row)

SELECT lookup3be(-12345, 123);
 lookup3be  
------------
 1268105482
(1 row)

-- Test consistency (same input should give same hash)
SELECT lookup3be('consistent test') = lookup3be('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3be('initval test', 1) != lookup3be('initval test', 2);
 ?column? 
----------
 t
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3be'
ORDER BY proname, proargtypes;
  proname  | provolatile | proisstrict 
-----------+-------------+-------------
 lookup3be | i           | t
 lookup3be | i           | t
 lookup3be | i           | t
 lookup3be | i           | t
 lookup3be | i           | t
 lookup3be | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)

